scripts/controllers/user_tools_controller.ts

```diff
@@ -0,0 +1,146 @@
+import { Controller } from '@hotwired/stimulus'
+
+class UserTools extends Controller {
+  static targets = [ "fullname", "popup", "spinner", "userbutton", "usertools" ]
+
+  initialized: Boolean
+  fullnameTarget: HTMLElement
+  popupTarget: HTMLElement
+  usertoolsTarget: HTMLElement
+  spinnerTarget: HTMLElement
+  userbuttonTarget: HTMLElement
+
+  initialize(): void {
+    this.initialized = false;
+  }
+
+  connect() {
+    // get the userinfo then fire a pageview (can't cache in the page)
+    fetch(
+      '/whoami.json',
+      {
+        method: 'GET',
+      }
+    )
+    .then(response => response.json())
+    .then(data => {
+      var userId = (data || {}).analyticsUserId;
+      if (userId) {
+        window['ga']('set', 'userId', userId)
+
+        // We don't use this feature
+        // if(window.location.pathname === '/') {
+        //   $(document).ready(function() {
+        //     personalizeHomepage(userId)
+        //   })
+        // }
```

Comment:
if you want to keep this commented-out blob here, probably move `personalizeHomepage` into this file too

(or just delete them both. they're always in git history if we need them)